### PR TITLE
Register DataPlugin for Development

### PR DESCRIPTION
**Pull request by jschumme**
# Justification
In addition to the normal export functionality, we want to provide the ability to register a DataPlugin inplace. This is particulary helpful for the development work flow which requires the script to not be embedded in the uri file. This PR adds another item to the right-click context menu:

image

# Implementation

- we export the uri to `os.tmpdir()` and open it. This only works when usireg is associated with the uri file-extension.
- when that fails, we provide an error message that the uri couldn't be registered but has been exported to `tmpdir`

# Testing
e2e-test added

### CODE REVIEW

File: src/commands.ts [hunk 1]
```diff
@@ -137,6 +124,35 @@ export async function exportPluginFromContextMenu(uri: vscode.Uri): Promise<void
...
+    const extensions = await readOrRequestFileExtensionConfig(uri);
+    if (!extensions) {
+        return;
+    }
+
+    await vscu.exportDataPlugin(scriptPath, extensions.toString(), `${exportPath}`, false);
+    const process = await open(exportPath);
+    process.on('exit', rc => {
+        const foundUsiReg = rc === 0;
+        if (!foundUsiReg) {
```
Comment (EvaTi) [inline src/commands.ts:140]: What happens if the Plugin is already registered? Is the currently registered version overwritten silently?

File: src/test/e2e/export-plugin.test.ts [hunk 2]
```diff
@@ -96,5 +99,91 @@ describe('Basic UI Tests', () => {
...
+        assert.notStrictEqual(undefined, menuItem);
+        await menuItem?.select();
+
+        // Check if uri got exported
+        const uriExists = fs.existsSync(exportPath);
+        assert.ok(uriExists);
+
+        // Check if file content is correct
+        const fileContent = fs.readFileSync(exportPath);
+        assert.ok(fileContent.indexOf('<filefilters extension="*.csv"') > -1);
```
Comment (EvaTi) [inline src/test/e2e/export-plugin.test.ts:179]: Does this also check whether the plugin has really been registered in storetypes.usiconfig? Looks like it only checks whether the uri file gets created correctly.

**Review by EvaTi — APPROVED** [review]

File: src/commands.ts (hunk 1 above)
Comment (jschumme) [reply src/commands.ts:140]: It actually behaves like a double-click on the uri file. It shows the usiReg.exe dialog, prompting whether you really wish to overwrite the existing Plugin. It's not done silently.

File: src/test/e2e/export-plugin.test.ts (hunk 2 above)
Comment (jschumme) [reply src/test/e2e/export-plugin.test.ts:179]: The e2e test will not be able to check a successful registration. It has no USI and no usiReg.exe available. It has to run on empty GitHub machines.

**Pull request merged**